src/github.com/getlantern/balancer/balancer.go

```diff
@@ -72,7 +72,7 @@ func (b *Balancer) OnRequest(req *http.Request) {
 // error.
 func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
 	now := time.Now()
-	lastDialTime := time.Unix(atomic.SwapInt64(&b.lastDialTime, now.Unix()), 0)
```

Comment:
Here's the bug that causes the test to fail randomly. Changing to `UnixNano` yields more reliable timing.